### PR TITLE
Merge from unstable to master

### PR DESCRIPTION
**Pull request by ericniebler**

### CODE REVIEW

File: test/async_scope_test.cpp [hunk 1]
```diff
@@ -90,9 +90,22 @@ struct async_scope_test : testing::Test {
...
     async_manual_reset_event evt;
 
-    scope.spawn_on(transform(just(), [&]() noexcept {
-      evt.set();
-    }), thread.get_scheduler());
+    scope.spawn_on(
+      thread.get_scheduler(),
+      transform(just(), [&]() noexcept {
+        evt.set();
+      }));
```
Comment (lewissbaker) [inline test/async_scope_test.cpp:97]: Can use `just_with()` here?

File: test/async_scope_test.cpp [hunk 2]
```diff
@@ -79,8 +80,7 @@ struct async_scope_test : testing::Test {
             return transform(just(), [&]() noexcept {
               executed = true;
             });
```
Comment (lewissbaker) [inline test/async_scope_test.cpp:82]: Can use `just_with()` here?

File: test/async_scope_test.cpp [hunk 3]
```diff
@@ -127,11 +146,11 @@ TEST_F(async_scope_test, work_spawned_in_correct_context) {
   async_manual_reset_event evt;
   std::thread::id id;
   scope.spawn_on(
+      thread.get_scheduler(),
       transform(just(), [&]{
         id = std::this_thread::get_id();
         evt.set();
-      }),
-      thread.get_scheduler());
+      }));
```
Comment (lewissbaker) [inline test/async_scope_test.cpp:153]: Can use `just_with()` here?

File: test/async_scope_test.cpp [hunk 4]
```diff
@@ -171,20 +190,28 @@ TEST_F(async_scope_test, lots_of_threads_works) {
...
+                [&](decr&) noexcept {
+                  return sequence(
+                      transform(
+                        just(),
+                        [&]() noexcept {
+                          auto prev = count.fetch_add(1, std::memory_order_relaxed);
+                          if (prev + 1 == maxCount) {
+                            evt2.set();
+                          }
+                        }),
```
Comment (lewissbaker) [inline test/async_scope_test.cpp:211]: Can use `just_with()` here?

File: doc/api_reference.md [hunk 5]
```diff
@@ -1019,8 +1043,14 @@ namespace unifex
     // deallocated without being started.
     void spawn(sender);
 
-    // Implemented as spawn(on(sender, scheduler)).
-    void spawn(sender, scheduler);
+    // Implemented as spawn(on(scheduler, sender)).
+    void spawn_on(scheduler, sender);
+
+    // Implemented as spawn(transform(just(), invocable)).
+    void spawn_call(invocable);
```
[the remark this answers is not in the document]
Comment (lewissbaker) [reply doc/api_reference.md:1050]: In fact, I don't even see a `spawn_call()` method on the `async_scope` class.
Can we just remove this from the documentation?

**Review by lewissbaker — CHANGES_REQUESTED** [review]

File: doc/api_reference.md [hunk 6]
```diff
@@ -157,13 +157,35 @@ Returns a sender that completes synchronously by calling `set_done()`.
 
 Returns a sender that completes synchronously by calling `set_error()` with `e`.
 
+### `just_with(callable)`
```
[the remark this answers is not in the document]
Comment (ericniebler) [reply doc/api_reference.md:160]: `just_from` works. Should we also change `let_with`?

File: doc/api_reference.md [hunk 7]
```diff
@@ -157,13 +157,35 @@ Returns a sender that completes synchronously by calling `set_done()`.
...
+`just_with(callable)` is synonymous with `transform(just(), callable)`.
+
 ### `stop_if_requested()`
 
 Returns a sender that queries the receiver with `get_stop_token()`, tests the
 resulting stop token with `stop_requested()`. If the result is `true`, then
 the sender completes synchronously with `set_done()`. Otherwise, the sender
 completes synchronously by calling `set_value()` with no arguments.
 
+### `defer(callable)`
```
[the remark this answers is not in the document]
Comment (ericniebler) [reply doc/api_reference.md:179]: No opinions at the moment.

File: doc/api_reference.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (lewissbaker) [reply doc/api_reference.md]: Yes, I think `let_with` could become `let_from` to be consistent.

**Pull request merged**